tests/components/smlight/test_config_flow.py

```diff
@@ -0,0 +1,425 @@
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result.get("type") == FlowResultType.FORM
+    assert result.get("step_id") == "user"
+    assert result.get("errors") == {}
+
+    mock_info = Info.from_dict(loads(load_fixture("smlight/info.json")))
+    with (
+        patch(
+            "pysmlight.web.Api2.get_info",
+            return_value=mock_info,
+        ),
+        patch(
+            "pysmlight.web.Api2.check_auth_needed",
+            return_value=False,
+        ),
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "slzb-06.local",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "SLZB-06p7"
+    assert result["data"] == {
+        CONF_HOST: "slzb-06.local",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_zeroconf_flow(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the full zeroconf flow including authentication."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_ZEROCONF}, data=DISCOVERY_INFO
+    )
+
+    assert result.get("description_placeholders") == {"host": "SLZB-06.local"}
+    assert result.get("type") is FlowResultType.FORM
+    assert result.get("step_id") == "confirm_discovery"
+
+    mock_info = Info.from_dict(loads(load_fixture("smlight/info.json")))
+    with (
+        patch(
+            "pysmlight.web.Api2.get_info",
+            return_value=mock_info,
+        ),
+        patch(
+            "pysmlight.web.Api2.check_auth_needed",
+            return_value=True,
+        ),
+        patch(
+            "pysmlight.web.Api2.authenticate",
+        ),
+    ):
+        progress = hass.config_entries.flow.async_progress()
+        assert len(progress) == 1
+        assert progress[0].get("flow_id") == result["flow_id"]
+        assert progress[0]["context"].get("confirm_only") is True
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+        assert result2.get("type") is FlowResultType.FORM
+        assert result2.get("step_id") == "auth"
+
+        progress2 = hass.config_entries.flow.async_progress()
+        assert len(progress2) == 1
+        assert progress2[0].get("flow_id") == result["flow_id"]
+
+        result3 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_USERNAME: "test-user",
+                CONF_PASSWORD: "test-pass",
+            },
+        )
+
+        assert result3.get("type") is FlowResultType.CREATE_ENTRY
+        assert result3 == snapshot
```

Comment:
I don't really like snapshots in config_flow tests. Imo it's better to test the data directly

Reply:
ok, have dropped snapshot from here.